src/llama_stack_client/lib/agents/client_tool.py

```diff
@@ -64,3 +65,105 @@ def run(
         self, messages: List[Union[UserMessage, ToolResponseMessage]]
     ) -> List[Union[UserMessage, ToolResponseMessage]]:
         raise NotImplementedError
+
+
+class SingleMessageClientTool(ClientTool):
+    """
+    Helper class to handle custom tools that take a single message
+    Extending this class and implementing the `run_impl` method will
+    allow for the tool be called by the model and the necessary plumbing.
+    """
+
+    def run(self, messages: List[Union[UserMessage, ToolResponseMessage]]) -> List[ToolResponseMessage]:
```

Comment:
I wonder if we should change ClientTool to return an `Optional[ToolResponseMessage]` or a `ToolResponseMessage` -- when we had initially designed it, we had some uber general "pipeline" design in mind which doesn't seem to be needed in practice. 

Also why is `messages` not simply a `List[Message]`? Whatever flows into the tool is usually a `CompletionMessage`, right?

Reply:
I feel like this would be sufficient, then we can get rid of the `SingleMessageClientTool`

```
def run(self, messages: CompletionMessage) -> ToolResponseMessage
```